### PR TITLE
Plugin card attatchment

### PR DESCRIPTION
**Pull request by tejasj1718**
### What this PR does (required):
- implements the CardAttachment plugin.
- adds 4 Instance Methods

get - returns the files attached.
create - adds a list of files to the plugin
update - adds files to the list
delete - removes certain or all files.

How it works,
get - need no params
create - takes in files from body and rewrite flag from the query params.
update - takes in files and index from body, if the index is not provided, files are inserted at the end.
delete - takes in indices from body and deletes files or if not provided deletes everything. 

### Any information needed to test this feature (required):
- Cannot be tested at the moment.

### CODE REVIEW

File: server/models/plugins/cardAttachment.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+};
+
+cardAttachmentSchema.statics.delete = async (pluginId) => {
+  const plugin = await this.findbyId(pluginId);
+  plugin.files.map(async (file) => {
+    this.model("file").deleteOne(file);
+  });
+  plugin.files = [];
+  await plugin.save();
+};
```
Comment (ezeilo-su) [inline server/models/plugins/cardAttachment.js:39]: Ticket requires that you create instance methods, but the static methods that you've created. See [guide](https://mongoosejs.com/docs/guide.html#methods)

**Review by ezeilo-su — CHANGES_REQUESTED** [review]
Please change your static methods to instance ones.

File: server/models/plugins/cardAttachment.js (hunk 1 above)
Comment (tejasj1718) [reply server/models/plugins/cardAttachment.js]: fixed it

File: server/models/plugins/cardAttachment.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+const basePlugin = require("./basePlugin");
+
+const cardAttachmentSchema = new mongoose.Schema({
+  files: [{ type: ObjectId, ref: "file", default: [] }],
+});
+
+cardAttachmentSchema.methods.get = async (pluginId) => {
+  const plugin = await this.findById(pluginId);
+  return plugin.files;
+};
```
Comment (moffatethan) [inline server/models/plugins/cardAttachment.js:12]: So, we're actually going to be given a data structure such as `{ body, params, query}) into the arguments. And verify plugin will confirm if the plugin exists. The get methods can just return the files in a response like 

`{ response: 200, data: files }`

We want to keep our responses generic to ensure we are able to have the plugin a bit more functional in terms of incoming data validation.

Comment (moffatethan) [reply server/models/plugins/cardAttachment.js:12]: Let's make that change where applicable in the rest of the code.

**Review by moffatethan — CHANGES_REQUESTED** [review]
This looks great!

**Review by moffatethan — APPROVED** [review]

**Pull request merged**